cpu/sam0_common/include/periph_cpu_common.h

```diff
@@ -1095,6 +1095,78 @@ int rtc_tamper_register(gpio_t pin, gpio_flank_t flank);
 void rtc_tamper_enable(void);
 /** @} */
 
+/**
+ * @name sam0 User Configuration
+ *
+ *      The MCUs of this family contain a region of memory that is used to store
+ *      CPU configuration & calibration data.
+ *      It can be used to set persistent settings and has some additional space
+ *      to store user configuration data.
+ * @{
+ */
+
+/**
+ * @brief MCU configuration applied on start. The contents of this struct differ
+ *        between families.
+ */
+typedef struct sam0_aux_cfg_mapping nvm_user_page_t;
+
+/**
+ * @brief   Size of the free to use auxiliary area in the user page
+ */
+#ifdef FLASH_USER_PAGE_SIZE
+#define FLASH_USER_PAGE_AUX_SIZE (FLASH_USER_PAGE_SIZE - sizeof(nvm_user_page_t))
+#else
+#define FLASH_USER_PAGE_AUX_SIZE (AUX_PAGE_SIZE * AUX_NB_OF_PAGES - sizeof(nvm_user_page_t))
+#endif
+
+/**
+ * @brief   Reset the configuration area, apply a new configuration.
+ *
+ *
+ * @param   cfg     New MCU configuration, may be NULL.
+ *                  In that case, this will clear the remaining user area
+ *                  and apply the old configuration again.
+ */
+void sam0_flashpage_aux_reset(const nvm_user_page_t *cfg);
+
+/**
+ * @brief   Write data to the user configuration area.
+ *          This will write data to the remaining space after @see nvm_user_page_t
+ *          The size of this area depends on the MCU family used.
+ *
+ *          Will only write bits 1 -> 0. To reset bits to 1, call @see sam0_flashpage_aux_reset
+ *          This will reset the whole user area configuration.
+ *
+ *          Arbitrary data lengths and offsets are supported.
+ *
+ * @param   offset  Byte offset after @see nvm_user_page_t
+ *                  must be less than `FLASH_USER_PAGE_AUX_SIZE`
+ * @param   data    The data to write
+ * @param   len     Size of the data
+ */
+void sam0_flashpage_aux_write_raw(uint32_t offset, const void *data, size_t len);
```

Comment:
Do we need a new function for this, can't `flashpage_write_raw` be used? Isn't just the address different?

Reply:
If its just for padding reasons the padding logic might be able to be generalized, a lot of the arch could do the same, few CPU's can do writes from !`erase_state` - > `erase_state` without a page/sector erase.

Reply:
We need to use different commands to write to that flash area.
Internally they already share functions for the actual flash write logic (which means also normal flash writes can be unaligned now) but this is a different flash region and we really don't accidentally overwrite the CPU configuration, a separate function is more convenient as otherwise we would have to add a check for the address inside `flashpage_write_raw()`and change the behavior based on that.